cmd/test.go

```diff
@@ -0,0 +1,311 @@
+	}
+
+	return nil
+}
+
+// generateAllAssetFilenames generates asset filenames for all supported platforms
+func generateAllAssetFilenames(installSpec *spec.InstallSpec, version string) (map[string]string, error) {
+	assetFilenames := make(map[string]string)
+
+	// Get supported platforms, or use default common platforms
+	platforms := getSupportedPlatforms(installSpec)
+
+	// Generate filename for each platform
+	for _, platform := range platforms {
+		os := spec.PlatformOSString(platform.OS)
+		arch := spec.PlatformArchString(platform.Arch)
+		
+		if os == "" || arch == "" {
+			continue
+		}
+
+		// Create filename generator
+		generator := asset.NewFilenameGenerator(installSpec, version)
+		
+		// Generate filename for this platform
+		filename, err := generator.GenerateFilename(os, arch)
+		if err != nil {
+			log.WithError(err).Warnf("Failed to generate filename for %s/%s", os, arch)
+			continue
+		}
+
+		platformKey := fmt.Sprintf("%s/%s", os, arch)
+		assetFilenames[platformKey] = filename
+	}
+
+	return assetFilenames, nil
+}
+
+// getSupportedPlatforms returns the list of supported platforms
+func getSupportedPlatforms(installSpec *spec.InstallSpec) []spec.SupportedPlatformElement {
+	if len(installSpec.SupportedPlatforms) > 0 {
+		return installSpec.SupportedPlatforms
+	}
+
+	// Default to common platforms
+	return []spec.SupportedPlatformElement{
+		{OS: spec.SupportedPlatformOSPtr("linux"), Arch: spec.SupportedPlatformArchPtr("amd64")},
+		{OS: spec.SupportedPlatformOSPtr("linux"), Arch: spec.SupportedPlatformArchPtr("arm64")},
+		{OS: spec.SupportedPlatformOSPtr("darwin"), Arch: spec.SupportedPlatformArchPtr("amd64")},
+		{OS: spec.SupportedPlatformOSPtr("darwin"), Arch: spec.SupportedPlatformArchPtr("arm64")},
+		{OS: spec.SupportedPlatformOSPtr("windows"), Arch: spec.SupportedPlatformArchPtr("amd64")},
+		{OS: spec.SupportedPlatformOSPtr("windows"), Arch: spec.SupportedPlatformArchPtr("arm64")},
+	}
+}
+
+// displayAssetFilenames displays the generated asset filenames in a table format
+func displayAssetFilenames(assetFilenames map[string]string) {
+	if len(assetFilenames) == 0 {
+		fmt.Println("No asset filenames generated")
+		return
+	}
+
+	// Sort platforms for consistent output
+	platforms := make([]string, 0, len(assetFilenames))
+	for platform := range assetFilenames {
+		platforms = append(platforms, platform)
+	}
+	sort.Strings(platforms)
+
+	// Create table writer
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+	fmt.Fprintln(w, "PLATFORM\tASSET FILENAME")
+	fmt.Fprintln(w, "--------\t--------------")
+
+	for _, platform := range platforms {
+		filename := assetFilenames[platform]
+		fmt.Fprintf(w, "%s\t%s\n", platform, filename)
+	}
+
+	w.Flush()
+}
+
+// checkAssetsExist checks if the generated asset filenames exist in the GitHub release
+func checkAssetsExist(ctx context.Context, installSpec *spec.InstallSpec, version string, assetFilenames map[string]string) error {
```

Comment:
Instead of checking files one by one, get list of release assets once and compare against them. 

We should also display released assets which doesn't have matching entry.